exporter/exporter.go

```diff
@@ -157,7 +161,18 @@ func (e *Exporter) handler() http.Handler {
 		ctx := r.Context()
 
 		client := e.client
-		topologyInfo := e.topologyInfo
+
+		if err := e.loadTopologyInfo(ctx, client); err != nil {
```

Comment:
@percona-csalguero as far as I know we provide `global-conn-pool` flag to mongodb_exporter and it shouldn't affect anything. And based on discussion with @AlekSi if `global-conn-pool` is false then each request should be like separate world.

Reply:
It would be the right thing to do, but it doesn't look like we can do it right now.

I tried to run `make test-cluster` and `MONGODB_URI=mongodb://127.0.0.1:17001 go run main.go --compatible-mode --mongodb.global-conn-pool` locally on macOS and measure time – `curl http://127.0.0.1:9216/metrics` takes 4 seconds (!!!). That's insane given that we want to scrape exporter every second.

I then took a profile (added pprof.samples.cpu.001.pb.gz to the ticket). It shows that most of the time is spent on starting and waiting for external processes from https://github.com/shirou/gopsutil. Unless it is something macOS specific, we should not start external processes on every scrape.

Please try it locally on Linux – `go tool pprof -http=: http://127.0.0.1:9216/debug/pprof/profile` doesn't work on Carlos' machine for some reason.